packages/expo-modules-core/ios/Core/Promise.swift

```diff
@@ -4,6 +4,7 @@ public struct Promise: AnyArgument {
   public typealias ResolveClosure = (Any?) -> Void
   public typealias RejectClosure = (Exception) -> Void
 
+  internal let appContext: AppContext
```

Comment:
Could we convert it to a weak ref? We don't know what people will do with it, which could lead to leaks.

Reply:
Yeah, we can do. It's marked as `internal` so nobody outside of core should be able to set it but it's probably worth doing in case we cause leaks